include/zephyr/drivers/stepper.h

```diff
@@ -0,0 +1,395 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_STEPPER_H
+#define ZEPHYR_INCLUDE_STEPPER_H
+
+/** \addtogroup stepper
+ *  @{
+ */
+
+/**
+ * @file
+ * @brief Public API for Stepper Motor Controller
+ *
+ */
+
+#include <zephyr/kernel.h>
+#include <errno.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Stepper Motor micro step resolution options
+ */
+enum micro_step_resolution {
+	STEPPER_FULL_STEP,
+	STEPPER_MICRO_STEP_2,
+	STEPPER_MICRO_STEP_4,
+	STEPPER_MICRO_STEP_8,
+	STEPPER_MICRO_STEP_16,
+	STEPPER_MICRO_STEP_32,
+	STEPPER_MICRO_STEP_64,
+	STEPPER_MICRO_STEP_128,
+	STEPPER_MICRO_STEP_256,
+};
+
+enum stepper_direction {
+	STEPPER_DIRECTION_POSITIVE,
+	STEPPER_DIRECTION_NEGATIVE,
+};
+
+/**
+ * @typedef stepper_enable_t
+ * @brief enable or disable the stepper motor controller.
+ */
+typedef int32_t (*stepper_enable_t)(const struct device *dev, bool enable);
+
+/**
+ * @typedef stepper_move_t
+ * @brief Stepper Motor Controller shall relatively move the motor with given steps
+ */
+typedef int32_t (*stepper_move_t)(const struct device *dev, int32_t steps);
+
+/**
+ * @typedef stepper_set_max_velocity_t
+ * @brief Set the max velocity in steps per seconds.
+ */
+typedef int32_t (*stepper_set_max_velocity_t)(const struct device *dev, uint32_t steps_per_second);
+
+/**
+ * @typedef stepper_set_micro_step_res_t
+ * @brief Set the micro-step resolution
+ */
+typedef int32_t (*stepper_set_micro_step_res_t)(const struct device *dev,
+						enum micro_step_resolution resolution);
+
+/**
+ * @typedef stepper_set_actual_position_t
+ * @brief Set the actual a.k.a reference position of the stepper
+ */
+typedef int32_t (*stepper_set_actual_position_t)(const struct device *dev, const int32_t value);
+
+/**
+ * @typedef stepper_get_actual_position_t
+ * @brief Get the actual a.k.a reference position of the stepper
+ */
+typedef int32_t (*stepper_get_actual_position_t)(const struct device *dev, int32_t *value);
+
+/**
+ * @typedef stepper_set_target_position_t
+ * @brief Set the absolute target position of the stepper
+ */
+typedef int32_t (*stepper_set_target_position_t)(const struct device *dev, const int32_t value);
+
+/**
+ * @typedef stepper_is_moving_t
+ * @brief Is the target position fo the stepper reached
+ */
+typedef int32_t (*stepper_is_moving_t)(const struct device *dev, bool *is_moving);
+
+/**
+ * @typedef stepper_enable_constant_velocity_mode_t
+ * @brief Enable constant velocity mode for the stepper with a given velocity
+ */
+typedef int32_t (*stepper_enable_constant_velocity_mode_t)(const struct device *dev,
+							   const enum stepper_direction direction,
+							   const uint32_t value);
```

Comment:
There is a small difference, by setting the velocity it is not imperative that the motor shall be set into motion, hence I've selected enable instead of set

Reply:
> by setting the velocity it is not imperative that the motor shall be set into motion

This function as written implies that the motor will begin moving as soon as it is set to a non-zero value and the driver is globally enabled. If there is something that would stop it from starting to move, that should be documented.

Reply:
It is documented in the brief of the function and not the typedef. do let me know if that's not enough or something is missing your opinion